### PR TITLE
Fang develop branch

### PR DESCRIPTION
**Pull request by lmrodriguezr**

### CODE REVIEW

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -333,6 +333,46 @@ def create_reference
...
+    #put on the map 
+    @map_p = Hash.new
+    @map_p[:total] = total
+    @map_p[:inactive] = inactive
+    @map_p[:incomplete] = incomplete
+    @map_p[:complete] = complete
+    @map_p[:percentage] = p_complete
+    @map_p[:last_line] = last_line
+    @map_p[:last_five_lines] = last_five_lines
+    @map_p[:active] = active
```
Comment (lmrodriguezr) [inline app/controllers/projects_controller.rb:368]:
```suggestion
    @map_p = {
      total: total,
      inactive: inactive,
      incomplete: incomplete,
      complete: complete,
      percentage: p_complete,
      last_line: last_line,
      last_five_lines: last_five_lines,
      active: active
    }
```

File: app/views/projects/show.html.erb [hunk 2]
```diff
@@ -19,6 +19,158 @@
         it is not officially supported.
       </div>
     <% end %>
+
+    <h3 role="button" data-toggle="collapse" class="result-title collapsed" id="project-progress" href="#project-progress-container" aria-expanded="false" aria-controls="result-cnt-project_stats">
+    
+  
+    <% unless @project.miga.metadata[:arcgis].nil? %>
```
Comment (lmrodriguezr) [inline app/views/projects/show.html.erb:26]: This seems to be here by mistake. Please check

File: app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -333,6 +333,46 @@ def create_reference
...
+    total = status.size
+    inactive = status.count(:inactive)
+    complete = (total - incomplete) - inactive
+    p_incomplete = (incomplete.to_f * 100.0)/total
+    p_complete = 100.0 - p_incomplete
+
+    p_complete = p_complete.round(2)
+   
+    #reading outputfile
+    f = File.expand_path("daemon/MiGA\:#{project.miga.name}.output", project.miga.path)
```
Comment (lmrodriguezr) [inline app/controllers/projects_controller.rb:351]: There is no need to escape the colon, that should be handled below when accessing the file
```suggestion
    f = File.join(project.miga.path, 'daemon', "MiGA:#{project.miga.name}.output")
```

File: app/controllers/projects_controller.rb [hunk 4]
```diff
@@ -333,6 +333,46 @@ def create_reference
...
+    last_line = `tail -n 1 #{f}`
+    logger.info("last 1 : " + last_line)
+    last_five_lines = `tail -n 5 #{f}`
+    logger.info ("last 5 : " + last_five_lines)
```
Comment (lmrodriguezr) [inline app/controllers/projects_controller.rb:355]: That shell access wasn't safe, and would easily break (e.g., with a project path including spaces)
```suggestion
    last_five_lines = File.readlines(f)[-5, 5]
    last_line = last_five_lines.last
    logger.info("last 1 : " + last_line)
    logger.info ("last 5 : " + last_five_lines)
```

File: config/routes.rb [hunk 5]
```diff
@@ -1,4 +1,7 @@
 Rails.application.routes.draw do
+  #test_Fang
+  get 'projects/:id/progress' => 'projects#progress'
```
Comment (lmrodriguezr) [inline config/routes.rb:3]: So we can link to it easily in the future:
```suggestion
  get 'projects/:id/progress' => 'projects#progress', as: :project_progress
```

File: app/views/projects/show.html.erb (hunk 2 above)
Comment (FangYuan0059) [reply app/views/projects/show.html.erb:26]: The line 26 is for hiding the collapse button if the project has never been run.

Comment (lmrodriguezr) [reply app/views/projects/show.html.erb:26]: It seems this `arcgis` check is here by mistake, though. This check is to add a map at the top of a project when properly configured ([see this example](http://microbial-genomes.org/projects/GROS)).

Comment (FangYuan0059) [reply app/views/projects/show.html.erb:26]: Oh, I see. Thanks for the example. I misunderstood this code. I thought this was for hiding the collapse button if there were no result data to show. I will delete it.

**Pull request merged**